core/file.c

```diff
@@ -462,19 +457,13 @@ int check_git_sha(const char *filename, struct git_repository **git_p, const cha
 	return 1;
 }
 
-int parse_file(const char *filename)
+int parse_file_git(const char *loc, const char *branch, const char *user, bool is_remote, bool is_cloud)
 {
 	struct git_repository *git;
-	const char *branch = NULL;
 	char *current_sha = copy_string(saved_git_id);
-	struct memblock mem;
-	char *fmt;
-	int ret;
+	git = is_git_repository(loc, branch, user, is_remote, is_cloud);
 
-	git = is_git_repository(filename, &branch, NULL, false);
-	if (prefs.cloud_git_url &&
-	    strstr(filename, prefs.cloud_git_url)
-	    && git == dummy_git_repository) {
+	if (is_cloud && (!git || git == dummy_git_repository)) {
```

Comment:
same here - it's clear that you are turning the whole logic upside down. That may be the right thing to do, I just can't tell

Reply:
Yes, that was the idea - remove the parsing of URLs from the git machinery.